src/wallet/mod.rs

```diff
@@ -1361,23 +1365,44 @@ where
             return Ok((must_spend, vec![]));
         }
 
-        let satisfies_confirmed = match must_only_use_confirmed_tx {
-            true => {
-                let database = self.database.borrow();
-                may_spend
-                    .iter()
-                    .map(|u| {
-                        database
-                            .get_tx(&u.0.outpoint.txid, true)
-                            .map(|tx| match tx {
-                                None => false,
-                                Some(tx) => tx.confirmation_time.is_some(),
-                            })
+        let database = self.database.borrow();
+        let satisfies_confirmed = may_spend
+            .iter()
+            .map(|u| {
+                database
+                    .get_tx(&u.0.outpoint.txid, true)
+                    .map(|tx| match tx {
+                        // We don't have the tx in the db for some reason,
+                        // so we can't know for sure if it's mature or not.
+                        // We prefer not to spend it.
+                        None => false,
+                        Some(tx) => {
+                            // Whether the UTXO is mature and, if needed, confirmed
+                            let mut spendable = true;
+                            if tx
+                                .transaction
+                                .expect("We specifically ask for the transaction above")
+                                .is_coin_base()
+                            {
+                                if let Some(current_height) = current_height {
```

Comment:
If `current_height` is not explicitly provided, would it make sense to grab it from `Database::get_sync_time`?

Reply:
We actually always get the current height from last sync height here if not provided. 
https://github.com/bitcoindevkit/bdk/blob/063d51fd7587815c733f24365a4de02fed072903/src/wallet/mod.rs#L622-L630

So theoretically this `current_height` will never be `None` here I think.. So we can as well use an `u32` instead of `Option<u32>` in `preselect_utxos`.

Reply:
> So theoretically this current_height will never be None here I think..

It can still be None if we never synced before (`get_sync_time` would return `None`)

Reply:
Ah yes that makes sense.. When we haven't synced ever it will be None at that time only.. But can't we just translate that None to 0 in that case and get rid of `Option`s all together in this level?? Not sure if it helps too much, but can simplify the code a bit.

Reply:
I would keep the option here, it feels easier to understand to me

Reply:
Ya not a biggie.. Resolving this now..